src/AdoNetCore.AseClient/Internal/Handler/EnvChangeTokenHandler.cs

```diff
@@ -89,5 +80,29 @@ public void Handle(IToken token)
                     return;
             }
         }
+
+        private void ApplyNewEncoding(EnvironmentChangeToken.EnvironmentChange change)
+        {
+            string newCharset = change.NewValue ?? string.Empty;
+            try
+            {
+                if (newCharset.Equals(string.Empty))
+                {
+                    newCharset = _requestedCharset ?? string.Empty;
+                }
```

Comment:
It would be a bit more readable if this if statement was outside the try.. catch. And it can't throw.

Reply:
Fixed